frontend/src/component/admin/groups/GroupsList/GroupCard/GroupCardAvatars/NewGroupCardAvatars.tsx

```diff
@@ -42,12 +43,23 @@ interface IGroupCardAvatarsProps {
     }[];
     header?: ReactNode;
     avatarLimit?: number;
+    AvatarComponent?: typeof UserAvatar;
 }
 
 export const GroupCardAvatars = ({
+    AvatarComponent,
+    ...props
+}: IGroupCardAvatarsProps) => {
+    const Avatar = StyledAvatar(AvatarComponent ?? UserAvatar);
+
+    return <GroupCardAvatarsInner AvatarComponent={Avatar} {...props} />;
+};
+
+const GroupCardAvatarsInner = ({
```

Comment:
This is an interesting use case that I want to explain.

By allowing you to pass a custom avatar prop, you can add your own custom styling to the avatars used by the group avatar component. This is useful in this instance because we want the avatars to be smaller than the group card avatar size. 

By currying the `StyledAvatar` function and allowing you to partially apply it by passing in the component, we allow you to pass a component in that we then add more styles to. This feels pretty cool to me.

However, I had to extract it into its own component because when you moused over an avatar, that'd cause the component to do some recalculations, leading to a rerendering of the avatar. This had two side effects when trying to open the popover:
1. It would constantly retrigger the onMouseOver function.
2. Because it didn't have a target element anymore, the popover would be rendered in the top left of the screen as shown in this image: 
![image](https://github.com/user-attachments/assets/ae75b8a6-f877-4158-a845-9adccb85a0e9)

By extracting the Avatar styling to an outside function, we can get around that completely.